sdk/sdk-skeleton.go

```diff
@@ -0,0 +1,77 @@
+type Asset struct{}
+
+//// Methods directly on the SDK
+
+func NewChannel() {
+
+}
+
+func NewChannelListenerService() {
+
+}
+
+//// Channel State Machine
+
+type ChannelStateMachine struct {
+	Status                 string // waiting | active | closing | closed
+	Asset                  Asset
+	InitiatorEscrowAccount string
+	ResponderEscrowAccount string
+	InitiatorClaimAmount   int
+	ResponderClaimAmount   int
+	IsInitiator            bool
+	MyAccount              string
+	MyClaimAmount          int
+	OtherAccount           string
+	OtherClaimAmount       int
```

Comment:
We have two different perspectives in this type's field names:
- Initiator/Responder are objective.
- My/Other are subjective.

The differing perspectives are confusing and difficult to think about, at least for me.

I see a few ways we could represent this data. Ideally we pick only one, if possible:

1. Initiator / Responder, objective. The `IsInitiator` field would signal which the caller would be. This will result in `if`s in the code wherever the user is making an action to select the right escrow account, etc.
2. My / Other, subjective. This probably makes some of the functionality simpler because it operates immediately on the appropriate object. We'd still need an indicator for who is the initiator, for selecting the sequencing number.
3. ParticipantA / ParticipantB, objective, and disconnected from roles. Roles assigned.

It might be useful for us to group all the fields relating to a single participant into a Participant type.

Reply:
I'm initially leaning towards My/Other. I think this is the most intuitive for the user of the sdk. Eg. a user has many payment channels going at once, it would be hard to remember which ones they are an Initiator vs Responder of w/o having to check IsInitiator all the time. Once the channel is live it doesn't really matter who Initiator vs Responder is 

Also not having to add `if` statements everywhere is nice 😄

Reply:
Just thought of one more: Local/Remote. 

I think My / Other sounds good though. Let's use it.

There will still need to be ifs around IsIniator to select the appropriate sequencing account, but I think that's fine.

Reply:
I think the SDK should save both perspectives internally (my/other probably as functions instead of variables maybe?). This allows the SDK to be cognizant of the Initiator/Responder as it needs to deal with the details of the channel. However, we can decide what API to present to the user, and I agree am leaning to My/Other (or Local/Remote) since that's a better API for the SDK consumer

Reply:
sounds good, leaving both perspectives for now then. If we decide to change that later it can be in a PR